integrations/tensorflow/compiler/dialect/tf_strings/ir/ops.td

```diff
@@ -44,6 +44,36 @@ def TFStrings_ToStringOp : TFStrings_Op<"to_string"> {
     OpBuilderDAG<(ins "Value":$value)>];
 }
 
+def TFStrings_GatherOp : Op<TFStrings_Dialect, "gather"> {
+  let summary = [{gathers all the strings from a Tensor by index}];
```

Comment:
1. use quotes and capitalize the first word for consistency
2. Are these indices or IDs? args and description refer to IDs, not index? https://www.tensorflow.org/api_docs/python/tf/gather refers to indices

ID = identifier, ordering does not necessarily matter
Index = value in range [0, shape[axis]]

```suggestion
  let summary = "Gathers all the strings from a Tensor by IDs.";
```